tests/optimize/hyperopts/default_hyperopt.py

```diff
@@ -54,38 +54,57 @@ def populate_buy_trend(dataframe: DataFrame, metadata: dict) -> DataFrame:
             """
             Buy strategy Hyperopt will build and use.
             """
-            conditions = []
+            long_conditions = []
+            short_conditions = []
```

Comment:
First of all - to ensure compatibility (existing things cannot break because of short support)- please do not touch these files (this and the corresponding strategy file).

We will have to have a short test file too - but in my opinion, not a explicit hyperopt file (see above - deprectated interface) - but only a seperate parametrized short strategy.

Reply:
What is the legacy hyperopt and what is the new hyperopt? Also, if it's legacy, can we add a Deprecated warning or something, and could we do that for everything else that's legacy so that I don't write updates for any other legacy code? I don't see anything in there that indicates it's legacy code

I created a short hyperopt for now, which is just the changes I had for hyperopt, but reverted the regular hyperopt file back

Reply:
parametrized hyperopting (hyperoptable parameters in the strategy) are the new hyperopt method.

The old / legacy method is using a seperate file.
While we're not showing a deprecation warning (intentionally for now) - the documentation does mention this as [legacy hyperopt](https://www.freqtrade.io/en/stable/advanced-hyperopt/#legacy-hyperopt).